src/teleport.cpp

```diff
@@ -0,0 +1,92 @@
+#include "teleport.h"
+
+#include "character.h"
+#include "avatar.h"
+#include "creature.h"
+#include "player.h"
+#include "game.h"
+#include "map.h"
+#include "messages.h"
+#include "point.h"
+
+
+const efftype_id effect_teleglow( "teleglow" );
+
+bool teleport::teleport( Creature *const critter, int min_distance, int max_distance, bool safe,
```

Comment:
```suggestion
bool teleport::teleport( Creature &critter, int min_distance, int max_distance, bool safe,
```
That way callers *can* simply not supply a null pointer, and *must* supply a valid object. So you don't need to test for this inside the function.